File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/commands/arena/ArenaLeaveCommand.java
package com.dumbdogdiner.warrior.commands.arena;

import com.dumbdogdiner.warrior.Warrior;
import com.dumbdogdiner.warrior.api.translation.Constants;
import com.dumbdogdiner.warrior.user.User;
import com.dumbdogdiner.warrior.api.command.SubCommand;
import com.dumbdogdiner.warrior.api.sessions.LobbySession;
import com.dumbdogdiner.warrior.api.sessions.SessionType;
import com.dumbdogdiner.warrior.user.UserCache;
import com.dumbdogdiner.warrior.api.util.TranslationUtil;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.List;

public class ArenaLeaveCommand implements SubCommand {
    @Override
    public String getAlias() {
        return "leave";
    }

    @Override
    public String getSyntax() {
        return "/arena leave";
    }

    @Override
    public String getPermission() {
        return "warrior.arena.leave";
    }

    @Override
    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
        User user = UserCache.get(((Player)sender).getUniqueId());
        if(user.getSession().getType() == SessionType.GAME) {
            String msg = Warrior.getTranslator().translate(Constants.Lang.ARENA_LEFT, new HashMap<>() {
                {
                    put("arena", args[1]);
                }
            }, user);

            user.sendMessage(TranslationUtil.getPrefix() + msg);
            user.setSession(new LobbySession(user.getUserId()));
            if(user.isSpectating()) user.setSpectating(false);
        }
        return true;
    }

    @Override
    public List<String> getArguments(CommandSender sender, String[] arguments) {
        return null;
    }
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/commands/arena/ArenaLeaveCommand.kt
package com.dumbdogdiner.warrior.commands.arena

import com.dumbdogdiner.warrior.api.command.SubCommand
import com.dumbdogdiner.warrior.api.sessions.SessionType
import com.dumbdogdiner.warrior.Warrior
import java.util.HashMap
import com.dumbdogdiner.warrior.api.util.TranslationUtil
import com.dumbdogdiner.warrior.api.sessions.LobbySession
import com.dumbdogdiner.warrior.api.translation.Constants
import com.dumbdogdiner.warrior.user.User
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class ArenaLeaveCommand : SubCommand {
    override fun getAlias(): String {
        return "leave"
    }

    override fun getSyntax(): String {
        return "/arena leave"
    }

    override fun getPermission(): String {
        return "warrior.arena.leave"
    }

    override fun execute(sender: CommandSender, commandLabel: String, args: Array<String>): Boolean {
        val user: User = Warrior.userCache[(sender as Player).uniqueId]
        if (user.session!!.type == SessionType.GAME) {
            val msg: String = Warrior.translator.translate(Constants.Lang.ARENA_LEFT, object : HashMap<String, String>() {
                init {
                    put("arena", args[1])
                }
            }, user)
            user.sendMessage(TranslationUtil.getPrefix() + msg)
            user.setSession(LobbySession(user.uniqueId))
            if (user.isSpectating) user.isSpectating = false
        }
        return true
    }

    override fun getArguments(sender: CommandSender, arguments: Array<String>): List<String> {
        return listOf()
    }
}